src/prefect/server/api/server.py

```diff
@@ -186,6 +187,7 @@ def create_orion_api(
     """
     fast_api_app_kwargs = fast_api_app_kwargs or {}
     api_app = FastAPI(title=API_TITLE, **fast_api_app_kwargs)
+    api_app.add_middleware(GZipMiddleware)
```

Comment:
Is this something we just want to add to all API routes? I don't know if it'll get us much of a benefit for many cases. Do we do this in Cloud?

Reply:
It doesn't hurt - clients have full control over how data is returned so clients that can't/don't want to handle compression can just ignore the `Accept-Encoding` header. This is a boon for browser clients especially because they send that header and handle compression automatically. 

To your second question: no, not yet. But soon 😄

Reply:
Oh also it should be noted that the `GZipMiddleware` has a min compression size of 500 bytes so it'll ignore payloads smaller than that. We could bump that if we think it's not worth it for a larger response body, I'm just not sure what that magic number is